Trouble Compiling

I'm using Raylib 3.0 on Windows 10 to run code very similar to examples/core/core_basic_window.cpp. This is the code:

```
#include "C:\raylib-cpp-master\include\raylib-cpp.hpp"

void UpdateDrawFrame(void);     // Update and Draw one frame

int main(void)
{
    const int screenWidth = 800;
    const int screenHeight = 450;
    
    raylib::Window w(screenWidth, screenHeight, "My first Window");
    SetTargetFPS(60);
    while (!w.ShouldClose())
    {
        UpdateDrawFrame();
    }
    return 0;
}

void UpdateDrawFrame(void)
{
    BeginDrawing();
        ClearBackground(RAYWHITE);
        DrawText("Hello World", 190, 200, 20, LIGHTGRAY);
    EndDrawing();
}
```

except I haven't been able to get that to compile. It complains about raymath.h missing. The output can be viewed here: https://pastebin.com/nCC9LNx4. So I go to the headers in the include folder and edit the raymath include statements to the path of my raylib/ src/raymath.h file. From:
`#include "raymath.h" // NOLINT`
to:
`#include "C:\raylib\raylib\src\raymath.h" // NOLINT`
But then I got a bunch of errors about undeclared functions: https://pastebin.com/LSL0eRfx. Note that I tried with release 3.0, 3.1, as well as the latest version from the master branch

I tried downloading the master version of raylib and changing the notepad++ script `raylib_compile_execute` to use that path instead. This is what I'm using to compile:
```
echo > Setup required Environment
echo -------------------------------------
SET RAYLIB_PATH=C:\raylib-master\raylib-master
SET COMPILER_PATH=C:\raylib\mingw\bin
ENV_SET PATH=$(COMPILER_PATH)
SET CC=g++
SET CFLAGS=$(RAYLIB_PATH)\src\raylib.rc.data -s -static -Os -std=c++17 -Wall -Iexternal -DPLATFORM_DESKTOP
SET LDFLAGS=-lraylib -lopengl32 -lgdi32 -lwinmm
cd $(CURRENT_DIRECTORY)
echo
echo > Clean latest build
echo ------------------------
cmd /c IF EXIST $(NAME_PART).exe del /F $(NAME_PART).exe
echo
echo > Saving Current File
echo -------------------------
npp_save
echo
echo > Compile program
echo -----------------------
$(CC) -o $(NAME_PART).exe $(FILE_NAME) $(CFLAGS) $(LDFLAGS)
echo
echo > Reset Environment
echo --------------------------
ENV_UNSET PATH
echo
echo > Execute program
echo -----------------------
cmd /c IF EXIST $(NAME_PART).exe $(NAME_PART).exe
```
doing that still gives me the raymath error, and when I try defining `RAYLIB_CPP_NO_MATH` I get errors. I'm probably doing something wrong here as I'm new to compiling, but I'm not sure what it could be.
```
g++ -o hello_world.exe hello_world.cpp C:\raylib\raylib\src\raylib.rc.data -s -static -Os -std=c++17 -Wall -Iexternal -DPLATFORM_DESKTOP -DRAYLIB_CPP_NO_MATH -lraylib -lopengl32 -lgdi32 -lwinmm
Process started (PID=6872) >>>
In file included from C:\raylib-cpp-master\include\raylib-cpp.hpp:82,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Color.hpp: In constructor 'raylib::Color::Color(Vector3)':
C:/raylib-cpp-master/include/Color.hpp:59:32: error: could not convert 'hsv.Vector3::x' from 'float' to 'Vector3'
         set(::ColorFromHSV(hsv.x, hsv.y, hsv.z));
                            ~~~~^
C:/raylib-cpp-master/include/Color.hpp: In static member function 'static raylib::Color raylib::Color::FromHSV(float, float, float)':
C:/raylib-cpp-master/include/Color.hpp:63:40: error: could not convert 'hue' from 'float' to 'Vector3'
         ::Color color = ::ColorFromHSV(hue, saturation, value);
                                        ^~~
In file included from C:\raylib-cpp-master\include\raylib-cpp.hpp:84,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Font.hpp: In constructor 'raylib::Font::Font(const string&, const unsigned char*, int, int, int*, int)':
C:/raylib-cpp-master/include/Font.hpp:54:15: error: '::LoadFontFromMemory' has not been declared
         set(::LoadFontFromMemory(fileType.c_str(), fileData, dataSize, fontSize, fontChars,
               ^~~~~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Font.hpp:54:15: note: suggested alternative: 'LoadFontFromImage'
         set(::LoadFontFromMemory(fileType.c_str(), fileData, dataSize, fontSize, fontChars,
               ^~~~~~~~~~~~~~~~~~
               LoadFontFromImage
In file included from C:\raylib-cpp-master\include\raylib-cpp.hpp:86,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Image.hpp: In member function 'void raylib::Image::LoadAnim(const string&, int*)':
C:/raylib-cpp-master/include/Image.hpp:128:15: error: '::LoadImageAnim' has not been declared
         set(::LoadImageAnim(fileName.c_str(), frames));
               ^~~~~~~~~~~~~
C:/raylib-cpp-master/include/Image.hpp:128:15: note: suggested alternative: 'LoadImageRaw'
         set(::LoadImageAnim(fileName.c_str(), frames));
               ^~~~~~~~~~~~~
               LoadImageRaw
C:/raylib-cpp-master/include/Image.hpp: In member function 'void raylib::Image::LoadFromMemory(const string&, const unsigned char*, int)':
C:/raylib-cpp-master/include/Image.hpp:132:15: error: '::LoadImageFromMemory' has not been declared
         set(::LoadImageFromMemory(fileType.c_str(), fileData, dataSize));
               ^~~~~~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Image.hpp:132:15: note: suggested alternative: 'LoadFromMemory'
         set(::LoadImageFromMemory(fileType.c_str(), fileData, dataSize));
               ^~~~~~~~~~~~~~~~~~~
               LoadFromMemory
C:/raylib-cpp-master/include/Image.hpp: In member function 'Color* raylib::Image::GetPalette(int, int*)':
C:/raylib-cpp-master/include/Image.hpp:274:18: error: '::GetImagePalette' has not been declared
         return ::GetImagePalette(*this, maxPaletteSize, extractCount);
                  ^~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Image.hpp:274:18: note: suggested alternative: 'GetPalette'
         return ::GetImagePalette(*this, maxPaletteSize, extractCount);
                  ^~~~~~~~~~~~~~~
                  GetPalette
C:/raylib-cpp-master/include/Image.hpp: In member function 'raylib::Image& raylib::Image::DrawText(const string&, Vector2, int, Color)':
C:/raylib-cpp-master/include/Image.hpp:345:23: error: could not convert '(& text)->std::__cxx11::basic_string<char>::c_str()' from 'const char*' to 'Vector2'
             text.c_str(),
             ~~~~~~~~~~^~
C:/raylib-cpp-master/include/Image.hpp: In member function 'raylib::Image& raylib::Image::DrawText(const string&, int, int, int, Color)':
C:/raylib-cpp-master/include/Image.hpp:354:41: error: could not convert '(& text)->std::__cxx11::basic_string<char>::c_str()' from 'const char*' to 'Vector2'
         ::ImageDrawText(this, text.c_str(), x, y, fontSize, color);
                               ~~~~~~~~~~^~
C:/raylib-cpp-master/include/Image.hpp: In member function 'raylib::Image& raylib::Image::DrawText(Font, const string&, Vector2, float, float, Color)':
C:/raylib-cpp-master/include/Image.hpp:360:33: error: could not convert 'font' from 'Font' to 'Vector2'
         ::ImageDrawTextEx(this, font, text.c_str(), position, fontSize, spacing, tint);
                                 ^~~~
In file included from C:\raylib-cpp-master\include\raylib-cpp.hpp:89,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Mesh.hpp: In member function 'raylib::Mesh& raylib::Mesh::NormalsSmooth()':
C:/raylib-cpp-master/include/Mesh.hpp:92:11: error: '::MeshNormalsSmooth' has not been declared
         ::MeshNormalsSmooth(this);
           ^~~~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Mesh.hpp:92:11: note: suggested alternative: 'NormalsSmooth'
         ::MeshNormalsSmooth(this);
           ^~~~~~~~~~~~~~~~~
           NormalsSmooth
In file included from C:\raylib-cpp-master\include\raylib-cpp.hpp:92,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Mouse.hpp: In member function 'int raylib::Mouse::GetCursor() const':
C:/raylib-cpp-master/include/Mouse.hpp:93:18: error: '::GetMouseCursor' has not been declared
         return ::GetMouseCursor();
                  ^~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Mouse.hpp:93:18: note: suggested alternative: 'GetCursor'
         return ::GetMouseCursor();
                  ^~~~~~~~~~~~~~
                  GetCursor
C:/raylib-cpp-master/include/Mouse.hpp: In member function 'raylib::Mouse& raylib::Mouse::SetCursor(int)':
C:/raylib-cpp-master/include/Mouse.hpp:97:11: error: '::SetMouseCursor' has not been declared
         ::SetMouseCursor(cursor);
           ^~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Mouse.hpp:97:11: note: suggested alternative: 'SetCursor'
         ::SetMouseCursor(cursor);
           ^~~~~~~~~~~~~~
           SetCursor
In file included from C:/raylib-cpp-master/include/AudioDevice.hpp:29,
                 from C:\raylib-cpp-master\include\raylib-cpp.hpp:77,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Music.hpp: In member function 'bool raylib::Music::GetLooping() const':
C:/raylib-cpp-master/include/Music.hpp:49:33: error: 'looping' was not declared in this scope
     GETTERSETTER(bool, Looping, looping)
                                 ^~~~~~~
C:/raylib-cpp-master/include/raylib-cpp-utils.hpp:37:46: note: in definition of macro 'GETTERSETTER'
     inline type Get##method() const { return name; } \
                                              ^~~~
C:/raylib-cpp-master/include/Music.hpp:49:33: note: suggested alternative: 'long'
     GETTERSETTER(bool, Looping, looping)
                                 ^~~~~~~
C:/raylib-cpp-master/include/raylib-cpp-utils.hpp:37:46: note: in definition of macro 'GETTERSETTER'
     inline type Get##method() const { return name; } \
                                              ^~~~
C:/raylib-cpp-master/include/Music.hpp: In member function 'void raylib::Music::SetLooping(bool)':
C:/raylib-cpp-master/include/Music.hpp:49:33: error: 'looping' was not declared in this scope
     GETTERSETTER(bool, Looping, looping)
                                 ^~~~~~~
C:/raylib-cpp-master/include/raylib-cpp-utils.hpp:38:43: note: in definition of macro 'GETTERSETTER'
     inline void Set##method(type value) { name = value; }
                                           ^~~~
C:/raylib-cpp-master/include/Music.hpp:49:33: note: suggested alternative: 'long'
     GETTERSETTER(bool, Looping, looping)
                                 ^~~~~~~
C:/raylib-cpp-master/include/raylib-cpp-utils.hpp:38:43: note: in definition of macro 'GETTERSETTER'
     inline void Set##method(type value) { name = value; }
                                           ^~~~
In file included from C:\raylib-cpp-master\include\raylib-cpp.hpp:93,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Music.hpp: In member function 'void raylib::Music::set(Music)':
C:/raylib-cpp-master/include/Music.hpp:117:9: error: 'looping' was not declared in this scope
         looping = music.looping;
         ^~~~~~~
C:/raylib-cpp-master/include/Music.hpp:117:9: note: suggested alternative: 'long'
         looping = music.looping;
         ^~~~~~~
         long
C:/raylib-cpp-master/include/Music.hpp:117:25: error: 'Music' {aka 'struct Music'} has no member named 'looping'; did you mean 'loopCount'?
         looping = music.looping;
                         ^~~~~~~
                         loopCount
In file included from C:/raylib-cpp-master/include/Shader.hpp:32,
                 from C:\raylib-cpp-master\include\raylib-cpp.hpp:98,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Texture.hpp: In member function 'raylib::Texture& raylib::Texture::UpdateRec(Rectangle, const void*)':
C:/raylib-cpp-master/include/Texture.hpp:92:9: error: 'UpdateTextureRec' was not declared in this scope
         UpdateTextureRec(*this, rec, pixels);
         ^~~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Texture.hpp:92:9: note: suggested alternative: 'UpdateTexture'
         UpdateTextureRec(*this, rec, pixels);
         ^~~~~~~~~~~~~~~~
         UpdateTexture
C:/raylib-cpp-master/include/Texture.hpp: In member function 'raylib::Texture& raylib::Texture::DrawTiled(raylib::Rectangle, raylib::Rectangle, raylib::Vector2, float, float, raylib::Color)':
C:/raylib-cpp-master/include/Texture.hpp:166:11: error: '::DrawTextureTiled' has not been declared
         ::DrawTextureTiled(*this, sourceRec, destRec, origin, rotation, scale, tint);
           ^~~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Texture.hpp:166:11: note: suggested alternative: 'DrawTextureQuad'
         ::DrawTextureTiled(*this, sourceRec, destRec, origin, rotation, scale, tint);
           ^~~~~~~~~~~~~~~~
           DrawTextureQuad
In file included from C:\raylib-cpp-master\include\raylib-cpp.hpp:98,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Shader.hpp: In member function 'int raylib::Shader::GetLocationAttrib(const string&) const':
C:/raylib-cpp-master/include/Shader.hpp:94:18: error: '::GetShaderLocationAttrib' has not been declared
         return ::GetShaderLocationAttrib(*this, attribName.c_str());
                  ^~~~~~~~~~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Shader.hpp:94:18: note: suggested alternative: 'GetLocationAttrib'
         return ::GetShaderLocationAttrib(*this, attribName.c_str());
                  ^~~~~~~~~~~~~~~~~~~~~~~
                  GetLocationAttrib
C:/raylib-cpp-master/include/Shader.hpp: In member function 'TextureCubemap raylib::Shader::GenTextureCubemap(raylib::Texture2D, int, int)':
C:/raylib-cpp-master/include/Shader.hpp:127:65: error: too many arguments to function 'Texture2D GenTextureCubemap(Shader, Texture2D, int)'
         return ::GenTextureCubemap(*this, panorama, size, format);
                                                                 ^
In file included from C:/raylib-cpp-master/include/raylib.hpp:31,
                 from C:/raylib-cpp-master/include/AudioDevice.hpp:28,
                 from C:\raylib-cpp-master\include\raylib-cpp.hpp:77,
                 from hello_world.cpp:2:
C:/raylib/mingw/i686-w64-mingw32/include/raylib.h:1370:17: note: declared here
 RLAPI Texture2D GenTextureCubemap(Shader shader, Texture2D map, int size);          // Generate cubemap texture from 2D texture
                 ^~~~~~~~~~~~~~~~~
In file included from C:\raylib-cpp-master\include\raylib-cpp.hpp:105,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Wave.hpp: In constructor 'raylib::Wave::Wave(const string&, const unsigned char*, int)':
C:/raylib-cpp-master/include/Wave.hpp:56:13: error: 'LoadWaveFromMemory' was not declared in this scope
         set(LoadWaveFromMemory(fileType.c_str(), fileData, dataSize));
             ^~~~~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Wave.hpp:56:13: note: suggested alternative: 'LoadModelFromMesh'
         set(LoadWaveFromMemory(fileType.c_str(), fileData, dataSize));
             ^~~~~~~~~~~~~~~~~~
             LoadModelFromMesh
In file included from C:\raylib-cpp-master\include\raylib-cpp.hpp:106,
                 from hello_world.cpp:2:
C:/raylib-cpp-master/include/Window.hpp: In member function 'bool raylib::Window::IsCursorOnScreen()':
C:/raylib-cpp-master/include/Window.hpp:82:18: error: '::IsCursorOnScreen' has not been declared
         return ::IsCursorOnScreen();
                  ^~~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Window.hpp:82:18: note: suggested alternative: 'IsCursorHidden'
         return ::IsCursorOnScreen();
                  ^~~~~~~~~~~~~~~~
                  IsCursorHidden
C:/raylib-cpp-master/include/Window.hpp: In member function 'bool raylib::Window::IsFocused() const':
C:/raylib-cpp-master/include/Window.hpp:103:18: error: '::IsWindowFocused' has not been declared
         return ::IsWindowFocused();
                  ^~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Window.hpp:103:18: note: suggested alternative: 'IsWindowResized'
         return ::IsWindowFocused();
                  ^~~~~~~~~~~~~~~
                  IsWindowResized
C:/raylib-cpp-master/include/Window.hpp: In member function 'raylib::Vector2 raylib::Window::GetScaleDPI() const':
C:/raylib-cpp-master/include/Window.hpp:241:18: error: '::GetWindowScaleDPI' has not been declared
         return ::GetWindowScaleDPI();
                  ^~~~~~~~~~~~~~~~~
C:/raylib-cpp-master/include/Window.hpp:241:18: note: suggested alternative: 'GetScaleDPI'
         return ::GetWindowScaleDPI();
                  ^~~~~~~~~~~~~~~~~
                  GetScaleDPI
<<< Process finished (PID=6872). (Exit code 1)
```

--- a/README.md
+++ b/README.md
@@ -206,6 +206,7 @@ raylib::Vector2 newDirection = direction.Rotate(30);
 The [projects directory](projects) includes some starter templates...
 
 - [CMake template](projects/CMake)
+- [Make template](projects/Make)
 
 If there's a project template you would like to see added, feel free to [make an issue](https://github.com/RobLoach/raylib-cpp/issues) and we can add it in.
 

--- a/projects/Make/README.md
+++ b/projects/Make/README.md
@@ -13,7 +13,7 @@ The [Raylib C++ Starter Kit](https://github.com/CapsCollective/raylib-cpp-starte
     ```
     cd raylib-cpp-starter
 
-    # Linux
+    # Linux & macOS
     make setup
     make
 

--- a/projects/Make/README.md
+++ b/projects/Make/README.md
@@ -1,0 +1,31 @@
+# raylib-cpp Make Example Project
+
+The [Raylib C++ Starter Kit](https://github.com/CapsCollective/raylib-cpp-starter), put together by [CapsCollective](https://caps-collective.itch.io), uses [`make`](https://www.gnu.org/software/make/) to compile a raylib-cpp starter project.
+
+## Build
+
+1. Check out the repository
+    ```
+    git clone https://github.com/CapsCollective/raylib-cpp-starter.git
+    ```
+
+2. Build the project
+    ```
+    cd raylib-cpp-starter
+
+    # Linux
+    make setup
+    make
+
+    # Windows
+    mingw32-make setup
+    mingw32-make
+    ```
+
+## Documentation
+
+See the [Raylib C++ Starter project](https://github.com/CapsCollective/raylib-cpp-starter) for more documentation.
+
+## Credits
+
+- [CapsCollective](https://caps-collective.itch.io)

